packages/storage/amplify_storage_s3_dart/lib/src/exception/s3_storage_exception.dart

```diff
@@ -22,6 +22,12 @@ const s3ControllableOperationCanceledException =
       ' exception allows you to take further action when an operation is canceled.',
 );
 
+/// The exception thrown when attempt to use acceleration endpoint with a bucket
+/// that has dots in its name.
+const accelerateEndpointUnusable = UnknownException(
```

Comment:
Do we have a service exception class? `UnknownException` doesn't seem correct

Reply:
There is no service exception for this, as `s3-accelerate` subdomain within a path style URL is unresolvable, in this case the API call returns a `NetworkException` which doesn't describe the actual issue. In addition, in the use case of `getUrl`, we need to detect this exception before an end user actually hits the non-working URL, where `NetworkException` is out of context.

Hence, we need to throw an exception before the non-working URL gets generated and used.

`UnknownException` implements `StorageException` so it can be caught by `on StorageException`, I think it's sufficient for this case.

Reply:
`ConfigurationError` is appropriate to be thrown when calling `plugin.configure`. 

`accelerateEndpointUnusable` will be thrown when calling `getUrl`, upload and download APIs (where `useAccelerateEndpoint` option is determinable). Excepting to catch only `StorageException` while calling APIs should be more straightforward without worry about a sorta unrelated exception. What do you think?

Reply:
Right, my point is it shouldn't be caught because it will always fail and the only recourse is to correct their usage of the API

Reply:
Ahh I got what you meant. Yea making the exception failing hard is clearer.